front-end-poc/src/ModelLookup/ModelDiagram.tsx

```diff
@@ -1,21 +1,60 @@
-import React from 'react';
-import { IModelVersion } from "./types"
+import React, { useEffect } from 'react';
+import ReactDOM from 'react-dom';
 import { LinearRegressionViewer, example3 } from "@manstis/pmml-linear-model-viewer"
+import { getModelDetail } from "../Shared/api/audit.api";
 
-const ModelDiagram = (props: { selectedVersion: IModelVersion }) => {
-  const { selectedVersion } = props;
-  const editorUrl = "https://kiegroup.github.io/kogito-online/?file=https://raw.githubusercontent.com/kiegroup/kogito-tooling/master/packages/online-editor/static/samples/sample.dmn#/editor/dmn";
-  const kogitoIframe = () => {
-    return { __html: `<iframe src=${editorUrl} data-key="${selectedVersion.version}"></iframe>` };
-  };
+interface Props {
+  id: string;
+}
+
+const ModelDiagram = (props: Props) => {
+
+  function embedDMNEditor(xml: string) {
+    const editorUrl = "https://kiegroup.github.io/kogito-online/?file=https://raw.githubusercontent.com/kiegroup/kogito-tooling/master/packages/online-editor/static/samples/sample.dmn#/editor/dmn";
+    const kogitoIframe = () => {
+      return { __html: `<iframe src=${editorUrl} data-key="${id}"></iframe>` };
+    };
+    ReactDOM.render(
+      <div className="model-diagram__iframe" dangerouslySetInnerHTML={kogitoIframe()} />,
+      document.getElementById("model-holder")
+    );
+  }
+
+  function embedPMMLEditor(xml: string) {
+    ReactDOM.render(
+      <LinearRegressionViewer xml={example3} />,
+      document.getElementById("model-holder")
+    );
+  }
+
+  const { id } = props;
+
+  useEffect(() => {
+    let didMount = true;
+    getModelDetail(id)
+      .then(response => {
+        if (didMount) {
+          const modelType: string = response.data[0].modelType;
+          const xml: string = response.data[0].xml;
+
+          switch (modelType) {
```

Comment:
@kelvah Is there a better way to handle embedding the correct viewer? I return a _place holder_ DOM element then use `ReactDOM.render(..)` to set the content of the placeholder once the model type and XML have been retrieved from the endpoint.

Reply:
@manstis The embed functions could return the `jsx` of each viewer, you could store it in a `viewer` variable inside the switch, and then use it in the return function of `ModelDiagram`, like `{viewer}`.
You might need to manage the state of the `viewer` with `useState`, given that you are manipulating it inside 'useEffect`.

Reply:
Great. Done.. much nicer. I also set the initial state to PatternFly's `<Spinner>`.